api/parts/mgmt/cms.js

```diff
@@ -0,0 +1,204 @@
+            return false;
+        }
+        if (results) {
+            transformAndStoreData(params, results, function(err1, storedResults) {
+                if (err1) {
+                    common.returnMessage(params, 500, 'An error occured while storing entries in DB: ' + err1);
+                    return false;
+                }
+                if (storedResults) {
+                    common.returnOutput(params, storedResults);
+                    return true;
+                }
+            });
+        }
+    });
+}
+
+/**
+* Get entries for a given API ID
+* @param {params} params - params object
+* @returns {boolean} true
+**/
+cmsApi.getEntries = function(params) {
+
+    if (!params.qstring._id || AVAILABLE_API_IDS.indexOf(params.qstring._id) === -1) {
+        common.returnMessage(params, 400, 'Missing or incorrect API _id parameter');
+        return false;
+    }
+
+    //WIP
+    var query = {'$or': [ {'_id': {'$regex': `^${params.qstring._id}`}}, {'_id': `${params.qstring._id}_meta`} ]};
+    if (params.qstring.query) {
+        for (var key in params.qstring.query) {
+            query[key] = params.qstring.query[key];
+        }
+    }
+
+    common.db.collection('cms_cache').find(query).toArray(function(err, entries) {
+        if (err) {
+            common.returnMessage(params, 500, 'An error occured while fetching CMS entries from DB: ' + err);
+            return false;
+        }
+        if (!entries || entries.length === 0) {
+            //No entries, fetch them
+            syncCMSDataToDB(params);
+        }
+        else {
+            const updateInterval = 24 * 60 * 60 * 1000;
+            const timeDifference = Date.now() - entries[0].lu;
+
+            if (entries.length === 1 && entries[0]._id === `${params.qstring._id}_meta`) {
+                //Only meta entry, check if it's time to update
+                if (timeDifference >= updateInterval) {
+                    syncCMSDataToDB(params);
+                }
+                else {
+                    common.returnOutput(params, []);
+                    return true;
+                }
+            }
+
+            //Entries exist, check if it's time to update
+            if (entries.length > 1 && timeDifference >= updateInterval) {
+                syncCMSDataToDB(params);
+            }
+            else {
+                common.returnOutput(params, entries || []);
+                return true;
+            }
+        }
+
+    });
+};
+
+/**
+* Clear cache for all API IDs
+* @param {params} params - params object
+**/
+cmsApi.clearCache = function(params) {
+    var query = {};
+    if (params.qstring._id) {
+        query = {'_id': {'$regex': `^${params.qstring._id}`}};
+    }
+    common.db.collection('cms_cache').deleteMany(query, function(err) {
+        if (err) {
+            common.returnMessage(params, 500, 'An error occured while clearing CMS cache: ' + err);
```

Comment:
You have to put in return or  do if....else.
It does not stop execution.
On error it will run line 198
And then line 200
Then you will have error. (Because you can't return output twice)